planet/src/main/java/com/google/planet/data/ItineraryGenerator.java

```diff
@@ -69,12 +81,16 @@
                 DistanceMatrixRow currentRow = distanceMatrix.rows[i];
                 for (int j = 0; j < currentRow.elements.length; j++) {
                     DistanceMatrixElement currentElement = currentRow.elements[j];
-                    travelTimeGraph[i][j] = currentElement.duration;
+                    if (currentElement.duration != null) {
+                        travelTimeGraph[i][j] = currentElement.duration;
+                    } else {
+                        errorMessage = "Status code 400 (BAD REQUEST): Oops, one of your addresses is invalid, please use a valid address.";
+                    }
                 }
             }
             return travelTimeGraph;
         } catch (Exception e) {
-            System.out.println(e);
+            errorMessage = "Status code 400 (BAD REQUEST): Oops, one of your addresses is invalid, please use a valid address.";
```

Comment:
nit: you don't need the "Status code 400 (BAD REQUEST)" in the string. Just the error message is enough.

Reply:
fixed!